examples/Cargo.toml

```diff
@@ -332,19 +384,24 @@ rand = { version = "0.8", optional = true }
 serde = { version = "1.0", features = ["derive"], optional = true }
 serde_json = { version = "1.0", optional = true }
 tracing = { version = "0.1.16", optional = true }
-tracing-subscriber = { version = "0.3", features = ["tracing-log", "fmt"], optional = true }
+tracing-subscriber = { version = "0.3", features = [
+    "tracing-log",
+    "fmt",
+], optional = true }
 prost-types = { version = "0.12", optional = true }
-http = { version = "0.2", optional = true }
-http-body = { version = "0.4.2", optional = true }
-hyper = { version = "0.14", optional = true }
-warp = { version = "0.3.4", default-features = false, optional = true }
+http = { version = "1", optional = true }
+http-body = { version = "1", optional = true }
+http-body-util = { version = "0.1", optional = true }
+hyper = { version = "1", optional = true }
+hyper-util = { version = "0.1", optional = true }
+# warp = { version = "0.3.4", default-features = false, optional = true }
 listenfd = { version = "1.0", optional = true }
 bytes = { version = "1", optional = true }
 h2 = { version = "0.3", optional = true }
-tokio-rustls = { version = "0.24.0", optional = true }
-hyper-rustls = { version = "0.24.0", features = ["http2"], optional = true }
-rustls-pemfile = { version = "1", optional = true }
-tower-http = { version = "0.4", optional = true }
+tokio-rustls = { version = "0.26.0", optional = true }
+hyper-rustls = { version = "0.27.0", features = ["http2"], optional = true }
```

Comment:
It would be nice if there were feature flags to switch between the crypto backends.  
Defaulting to `aws-lc-rs` pulls in a cmake, nasm requirement, impacts platform support, etc.  
Trading that off for FIPS compliance isn't really needed in most cases.

Reply:
Which crypto provider `tonic` should default to is up to the maintainers, I guess.  
But `reqwest` seems to be going the route of sticking to `ring` as the default crypto backend.

Reply:
I think that is a good idea, but I don't think it should be included in this PR - this PR is already complicated as is. We can create an issue and work on multi-backend support separately once this lands.

Reply:
Yeah, definitely. I just felt like the comment fit best here for the time being since the PR isn't merged yet and I probably would have forgotten about it if I hadn't written it down